financial-templates-lib/clients/ExpiringMultiPartyClient.js

```diff
@@ -58,6 +66,15 @@ class ExpiringMultiPartyClient {
   // Whether the given undisputed `liquidation` (`getUndisputedLiquidations` returns an array of `liquidation`s) is disputable.
   // `tokenRedemptionValue` should be the redemption value at `liquidation.time`.
   isDisputable = (liquidation, tokenRedemptionValue) => {
+    if (!liquidation || !liquidation.numTokens || !liquidation.amountCollateral || !tokenRedemptionValue) {
```

Comment:
High-level question: do we need to check for invalid values at every level of the stack? Another approach might be to decide where these checks should be performed and perform them exclusively at that level to avoid having redundant checks that bloat the codebase. For functions that *don't* validate their inputs, we can just document that it's a precondition that none of the required parameters are `undefined` or `null`.

For instance, in this case, it may make sense to assume that the caller guarantees that the inputs are valid.

Reply:
I think its impractical to have to check for invalid values at every level of the stack. Our general style has been to check for invalid values at the highest level of the stack, but the disputer calls `isDisputable` like so:


```
const disputeableLiquidations = undisputedLiquidations.filter(
      liquidation =>
        this.empClient.isDisputable(
          liquidation,
          this.priceFeed.getHistoricalPrice(parseInt(liquidation.liquidationTime.toString()))
        ) && this.empClient.getLastUpdateTime() >= Number(liquidation.liquidationTime) + this.disputeDelay
    );
```

We would therefore need to change this call by first passing through `undisputedLiquidations` and making sure we can get a price at each liquidation time, and then running the filter

Reply:
Actually there's an easier fix: just add a check `this.priceFeed.getHistoricalPrice(timestamp)` to the `&&` conditional

Reply:
Yep, that works. If you wanted to print a warning log message, you can expand the filter function and print a log message whenever `this.priceFeed.getHistoricalPrice(timestamp)` doesn't return a price.

Reply:
Yup I did this